crates/telio-traversal/src/upgrade_sync.rs

```diff
@@ -43,6 +51,20 @@ pub struct UpgradeRequest {
     pub session: Session,
 }
 
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy)]
+enum Direction<T: Debug + PartialEq + Eq + Hash + Clone + Copy> {
+    Sent(T),
+    Received(T),
+}
```

Comment:
Maybe `Directed`? Maybe you could add some comment why do we need it?

Reply:
I'm not sure about the name change - I feel that types should generally be nouns, not verbs. But for sure a comment is needed here! :)